multi: add address version to database

guggero opened the pull request:
With the recent PR https://github.com/lightninglabs/taproot-assets/pull/501 we changed the asset version within an address into the address' version itself. We still need the asset version in the address as well though, so this commit restores it, including the database storage.

EDIT: This re-orders all taproot address TLV field values to get them nicely incrementing. I'll follow up with a PR to the BIP.

ffranr commented:
@guggero I don't think the asset version field was removed in https://github.com/lightninglabs/taproot-assets/pull/501/files#diff-ccfff66eef18bce3fec0f292a24a84b7163361f1af5090d00b6139fdf8115479R102

ffranr reviewed (commented):
@guggero I don't think the asset version field was removed in https://github.com/lightninglabs/taproot-assets/pull/501/files#diff-ccfff66eef18bce3fec0f292a24a84b7163361f1af5090d00b6139fdf8115479R102

We just added a new field for the Tap version. The AssetVersion field was not removed or replaced.

I can see that this PR bumps all of the Tap TLV values. I think we should note that in the commit.

guggero commented:
> @guggero I don't think the asset version field was removed in https://github.com/lightninglabs/taproot-assets/pull/501/files#diff-ccfff66eef18bce3fec0f292a24a84b7163361f1af5090d00b6139fdf8115479R102
> 
> We just added a new field for the Tap version. The AssetVersion field was not removed or replaced.
> 
> I can see that this PR bumps all of the Tap TLV values. I think we should note that in the commit.

Ah yes, you're right, sorry. But I guess we didn't end up storing the address version in the DB.

Updated the PR description to mention the TLV values.

ffranr commented:
> we didn't end up storing the address version in the DB

Yes, true!

ffranr commented:
@guggero I think you still need to update the commit message.

guggero commented:
> @guggero I think you still need to update the commit message.

Right, fixed.

ffranr reviewed (approved):
Looks good to me! :+1:

Roasbeef reviewed (approved):
LGTM 🦚
